src/lib/utils/should-publish.ts

```diff
@@ -0,0 +1,14 @@
+import Entry from '../entities/entry'
+
+export default function shouldPublish (shouldPublish: boolean | 'preserve', entry: Entry) {
+  if (shouldPublish === true) {
+    return true
+  } else if (shouldPublish === false) {
+    return false
+  } else if (shouldPublish === 'preserve') {
+    const hasUnpublishedChanges = entry.version - entry.publishedVersion > 2
```

Comment:
should it be `>=`? https://www.contentful.com/developers/docs/tutorials/general/determine-entry-asset-state/

Reply:
@phoebeschmidt thanks for pointing out!
I double checked, and in this case it's a bit different, at the moment we call this function to check, we already updated the version field. I added some explanation to the function and also renamed it to something not so general.